client/src/app/site/pages/meetings/pages/motions/modules/motion-poll/components/motion-poll-vote/motion-poll-vote.component.ts

```diff
@@ -58,32 +58,158 @@ export class MotionPollVoteComponent extends BasePollVoteComponent implements On
         this.cd.markForCheck();
     }
 
-    public getActionButtonClass(voteOption: VoteOption, user: ViewUser = this.user): string {
-        if (this.voteRequestData[user?.id]?.value === voteOption.vote) {
+    public getActionButtonClass(voteOption: VoteOption, id: number, user: ViewUser = this.user): string {
+        if (
+            this.voteRequestData[user?.id]?.value[id] === voteOption.vote ||
+            this.voteRequestData[user?.id]?.value[id] === 1
+        ) {
             return voteOption.css!;
         }
         return ``;
     }
 
-    public async saveVote(vote: VoteValue, optionId: Id, user: ViewUser = this.user): Promise<void> {
+    public get minVotes(): number {
+        return this.poll.min_votes_amount;
+    }
+
+    public saveVote(optionId: Id, vote: VoteValue, user: ViewUser = this.user): void {
+        if (!this.voteRequestData[user.id]) {
+            throw new Error(`The user for your voting request does not exist`);
+        }
+
+        if (this.isGlobalOptionSelected(user)) {
+            this.voteRequestData[user.id].value = {};
+        }
+
+        if (this.poll.isMethodY || this.poll.isMethodN) {
+            const maxVotesAmount = this.poll.max_votes_amount;
+            const tmpVoteRequest = this.poll.options
+                .map(option => option.id)
+                .reduce((o: any, n) => {
+                    o[n] = 0;
+                    if (maxVotesAmount === 1) {
+                        if (n === optionId && this.voteRequestData[user.id].value[n] !== 1) {
+                            o[n] = 1;
+                        }
+                    } else if ((n === optionId) !== (this.voteRequestData[user.id].value[n] === 1)) {
+                        o[n] = 1;
+                    }
+
+                    return o;
+                }, {});
+
+            // check if you can still vote
+            const countedVotes = Object.keys(tmpVoteRequest).filter(key => tmpVoteRequest[key]).length;
+            if (countedVotes <= maxVotesAmount) {
+                this.voteRequestData[user.id].value = tmpVoteRequest;
+            } else {
+                this.raiseError(
+                    this.translate.instant(`You reached the maximum amount of votes. Deselect somebody first.`)
+                );
+            }
```

Comment:
Delete this part, motion polls can only be method `YNA`, so only the else clause of this will ever happen